### PR TITLE
feat: add C implementation for `math/base/special/maxabs`

### LINKED ISSUE #1460: [RFC]: Add C implementation for `@stdlib/math/base/special/maxabs`
### Description

This RFC proposes adding native C implementation for [`@stdlib/math/base/special/maxabs`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/maxabs).

```c
double stdlib_base_maxabs( const double x, const double y );
```

### Related Issues

Related issues #1433, #1449.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Shashankss1205**
Resolves #1460 .

## Description

> What is the purpose of this pull request?

This pull request adds a C implementation for `@stdlib/math/base/special/maxabs`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1460

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/maxabs/README.md
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/README.md]:
```suggestion
#include <stdio.h>

int main( void ) {
```

File: lib/node_modules/@stdlib/math/base/special/maxabs/README.md
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/README.md]:
```suggestion
Copyright (c) 2018 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/maxabs/include/stdlib/math/base/special/maxabs.h
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/include/stdlib/math/base/special/maxabs.h]:
```suggestion
* Returns the maximum absolute value.
```

File: lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json]:
```suggestion
        "@stdlib/math/base/special/max"
```

File: lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json]:
```suggestion
        "@stdlib/math/base/special/max"
```

File: lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/manifest.json]:
```suggestion
        "@stdlib/math/base/special/max"
```

**Review by Pranavchiku — APPROVED** [review]
This is good, thanks @Shashankss1205 !

File: lib/node_modules/@stdlib/math/base/special/maxabs/benchmark/c/native/Makefile
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/maxabs/benchmark/c/native/Makefile]: Trailing newline.

File: lib/node_modules/@stdlib/math/base/special/maxabs/benchmark/c/native/benchmark.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/maxabs/benchmark/c/native/benchmark.c]: Trailing newlines here and everywhere.

File: lib/node_modules/@stdlib/math/base/special/maxabs/src/main.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/maxabs/src/main.c]: Before this can be merged, we need to merge https://github.com/stdlib-js/stdlib/pull/1459.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @Shashankss1205. Left a few comments. This PR is blocked until https://github.com/stdlib-js/stdlib/pull/1459 is merged, given that it is a dependency of this implementation.

File: lib/node_modules/@stdlib/math/base/special/maxabs/src/main.c
Comment (Shashankss1205) [reply lib/node_modules/@stdlib/math/base/special/maxabs/src/main.c]: I am committing changes to that PR as well in another 10 mins.

**Comment by Shashankss1205** [comment]
> Thanks for working on this @Shashankss1205. Left a few comments. This PR is blocked until #1459 is merged, given that it is a dependency of this implementation.

I have committed the changes. Please take a look at it and inform if there are any changes that are needed. @kgryte

**Comment by Planeshifter** [comment]
@Shashankss1205 Now that https://github.com/stdlib-js/stdlib/pull/1459 is merged, would you please either merge in the latest changes from `develop` or rebase your PR branch so that CI will pass? Otherwise, I only found that a bunch of the C files are still missing trailing newlines. Once resolved, this PR should be ready to merge. Thank you very much!

**Comment by Shashankss1205** [comment]
Now can You please review the PR @Planeshifter . I have made the corresponding changes.

File: lib/node_modules/@stdlib/math/base/special/maxabs/examples/c/example.c
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/maxabs/examples/c/example.c]:
```suggestion
int main( void ) {
	double x;
	double y;
	double v;
	int i;

	for ( i = 0; i < 100; i++ ) {
		x = ( ( (double)rand() / (double)RAND_MAX ) * 1000.0 ) - 500.0;
		y = ( ( (double)rand() / (double)RAND_MAX ) * 1000.0 ) - 500.0;
		v = stdlib_base_maxabs( x, y );
		printf( "x: %lf, y: %lf, maxabs(x, y): %lf\n", x, y, v );
	}
}
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @Shashankss1205 !

**Comment by Shashankss1205** [comment]
> LGTM. Thanks @Shashankss1205 !

Thank You!! Looking forward to working more here.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**
